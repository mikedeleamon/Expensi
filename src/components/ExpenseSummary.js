import numeral from 'numeral'
import React from "react";
import {connect} from 'react-redux'
import selectExpenses from '../selectors/expenses'
import getExpenseTotal from '../selectors/expenses-total'

 export const ExpsenseSummary = ({expenseCount , expenseTotal}) => {

    const expensePlural = expenseCount === 1 ? 'expense':'expenses'
    const isPlural = expenseCount === 1 ? 'is':'are'
    const formatTotal = numeral(expenseTotal/100).format('$0,0.00')
    return (
    <div>
        <h1>Your total amount of {expenseCount} {expensePlural} {isPlural} {formatTotal}</h1>
    </div>
    )
}

//set state for connect desired from the store
const mapStatetoProps = (state)=> {
    const visibleExpenses = selectExpenses(state.expenses,state.filters)
    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: getExpenseTotal(visibleExpenses)
    }
}

// //connect takes in state
// //needs provider already set up
 export default connect(mapStatetoProps)(ExpsenseSummary)
