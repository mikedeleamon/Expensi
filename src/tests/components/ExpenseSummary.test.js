import React from 'react'
import {shallow} from 'enzyme'
import ExpenseSummary from '../../components/ExpenseSummary'

test('should render Summary of 1 expense', ()=>{
    // const wrapper = shallow(<ExpenseSummary expenseCount = {1} expenseTotal = {5000}/>)
    // expect(wrapper).toMatchSnapshot()
})

// test('should render Summary of 2 expenses', ()=>{
//     const wrapper = shallow(<ExpenseSummary expenseCount = {2} expenseTotal = {10000}/>)
//     expect(wrapper).toMatchSnapshot()
// })