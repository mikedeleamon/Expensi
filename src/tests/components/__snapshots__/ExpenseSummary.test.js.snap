// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Summary of 1 expense 1`] = `
<div>
  <h1>
    Your total amount of 
    1
     
    expense
     
    is
     
    $50.00
  </h1>
</div>
`;

exports[`should render Summary of 2 expenses 1`] = `
<div>
  <h1>
    Your total amount of 
    2
     
    expenses
     
    are
     
    $100.00
  </h1>
</div>
`;
